expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/extension/OperationToRequestExtension.kt

```diff
@@ -0,0 +1,134 @@
+import com.expediagroup.sdk.core.http.Request
+import com.expediagroup.sdk.core.http.RequestBody
+import com.expediagroup.sdk.rest.trait.operation.ContentTypeTrait
+import com.expediagroup.sdk.rest.trait.operation.HeadersTrait
+import com.expediagroup.sdk.rest.trait.operation.OperationRequestBodyTrait
+import com.expediagroup.sdk.rest.trait.operation.OperationRequestTrait
+import com.expediagroup.sdk.rest.trait.operation.UrlPathTrait
+import com.expediagroup.sdk.rest.trait.operation.UrlQueryParamsTrait
+import com.fasterxml.jackson.databind.ObjectMapper
+import java.io.IOException
+import java.net.MalformedURLException
+import java.net.URL
+
+/**
+ * Extension function to parse an operation request into an SDK request.
+ *
+ * This function takes an operation request and converts it into an SDK request
+ * by setting the HTTP method, headers, body, and URL based on the traits of the operation.
+ *
+ * @param url the base server URL
+ * @param mapper Jackson object mapper used to serialize the request body
+ * @return the constructed SDK request
+ * @throws IllegalArgumentException if the request body is invalid
+ * @throws IllegalStateException if the HTTP method or URL is not set
+ * @throws IOException if an I/O error occurs
+ */
+internal fun OperationRequestTrait.parseRequest(
+    url: URL,
+    mapper: ObjectMapper
+): Request {
+    require(this.getHttpMethod().isNotBlank()) {
+        "Operation HTTP method must not be empty!"
+    }
+
+    val builder = Request.builder().method(this.parseMethod())
+
+    if (this is HeadersTrait && this.getHeaders().entries().isNotEmpty()) {
+        builder.headers(this.getHeaders())
+    }
+
+    if (this is OperationRequestBodyTrait<*>) {
+        builder.body(this.parseRequestBody(mapper))
+    }
+
+    builder.url(
+        if (this is UrlPathTrait && this.getUrlPath().isNotBlank()) {
+            this.parseURL(url)
+        } else {
+            url
+        }
+    )
+
+    return builder.build()
+}
+
+/**
+ * Extension function to parse the URL of an operation request.
+ *
+ * This function constructs the full URL by combining the base URL with the path and query parameters
+ * from the operation request.
+ *
+ * @param base the base URL
+ * @return the constructed URL
+ * @throws MalformedURLException if the constructed URL is invalid
+ */
+internal fun UrlPathTrait.parseURL(base: URL): URL =
+    URL(
+        StringBuilder().apply {
+            append(base.toString().trim('/'))
+
+            if (this@parseURL.getUrlPath().isNotBlank()) {
+                append(this@parseURL.getUrlPath())
+            }
+
+            if (this@parseURL is UrlQueryParamsTrait && this@parseURL.getUrlQueryParams().isNotEmpty()) {
+                append("?")
+                this@parseURL.getUrlQueryParams().forEach { (key, values) ->
+                    if (key.isBlank()) {
+                        return@forEach
+                    }
+                    values.forEach { value -> append("$key=$value&") }
```

Comment:
As we discussed, there are different formats for serializing params. This is only one format. I would suggest we leave parsing URLs to the product..

https://swagger.io/docs/specification/v3_0/serialization/